### PR TITLE
Ax 871 add checks to the frontend for displaying buttons

### PR DESCRIPTION
**Pull request by MichaelFrank0317**
# Description
Handle checks to the frontend for displaying buttons on mint, liquidity and trade

Fixes Jira Ticket # 871
https://athletex.atlassian.net/browse/AX-871

## Type of change
Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Is this a UI Change? If so please include screenshot of before and after states below
image
image
image
image
image

# How Has This Been Tested?
This has been tested on the UI.

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have assigned 2 reviewers to check my work
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/dialogs/mint/bloc/mint_dialog_event.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+abstract class MintDialogEvent extends Equatable {
+  @override
+  List<Object?> get props => [];
+}
+
+class OnAxAmountChanged extends MintDialogEvent {
+  OnAxAmountChanged({required this.axAmount});
+
+  final double axAmount;
```
Comment (Friendlygeckos) [inline lib/dialogs/mint/bloc/mint_dialog_event.dart:11]: @override
  List<Object?> get props => [axAmount];

File: lib/pages/athlete/components/athlete_mint_approve_button.dart [hunk 2]
```diff
@@ -96,6 +107,8 @@ class _AthleteMintApproveButtonState extends State<AthleteMintApproveButton> {
       ),
       child: TextButton(
         onPressed: () {
+          if (iswaitingApproval) return;
```
Comment (Friendlygeckos) [inline lib/pages/athlete/components/athlete_mint_approve_button.dart:110]: what is going on over here? Why do we need it?

File: lib/pages/scout/desktop_scout.dart [hunk 3]
```diff
@@ -816,6 +822,11 @@ class _DesktopScoutState extends State<DesktopScout> {
                   ),
                   child: TextButton(
                     onPressed: () {
+                      if (!controller.walletConnected) {
+                        context.showWalletWarningToast();
+                        return;
```
Comment (Friendlygeckos) [inline lib/pages/scout/desktop_scout.dart:827]: why are we returning here?

File: lib/pages/scout/dialogs/athlete_page_dialogs.dart [hunk 4]
```diff
@@ -1,2 +1,2 @@
-export 'mint_dialog.dart';
+export '../../../dialogs/mint/mint_dialog.dart';
```
Comment (Friendlygeckos) [inline lib/pages/scout/dialogs/athlete_page_dialogs.dart:1]: provide the absolute path

**Review by Friendlygeckos — CHANGES_REQUESTED** [review]
please grey out the button for the mint and add liquidity buttons. Look at the example for the trade page

**Review by ketonkss4 — CHANGES_REQUESTED** [review]
Rebase!~

**Pull request closed without merging**